webapp/platform/components/src/generic_modal/generic_modal.tsx

```diff
@@ -184,13 +206,24 @@ export class GenericModal extends React.PureComponent<Props, State> {
             </div>
         );
 
+        const locationClassMapping: Record<Required<Props>['modalLocation'], string> = {
```

Comment:
This is a pretty good solution to this mapping problem, but it makes me miss Java-style enums. They let you give each `ModalLocation` methods so that you could just call something like `this.props.modalLocation.getCssClass()`

Reply:
ok, so after investigating a little bit I found that TS native enums don’t support methods directly, so we’d need to implement a custom solution like a class or a mapping object to get to mimic this Java pattern. So, I think that with just three location values and simple string mapping this current solution is enough.

Reply:
Yeah, I've dug a bunch into that as well, and I couldn't find any way to mimic Java's behaviour. I definitely agree that this is a good solution even if I wish TS made it a bit easier to write